### ISSUE TITLE
Dirty checking for entities

### ISSUE THREAD

**Issue by dsnipe** [lotus/model#170]
Why not to implement dirty checking for entities?

Something like this:
```ruby
class Article
  include Lotus::Entity
  attributes :title
end

record = Article.new(title: 'My title')
article.changed? # true
article = ArticleRepository.create(record)
article.changed? # false

article.title = 'New title'
article.changed? # true
article.changed_attributes # {:title => 'My title'}
```
I think to put all in 'entity' for now. 
Maybe any thoughts about it?

**Comment by mvader** [comment]
I think it would be really cool to have that feature.

**Comment by jodosha** [comment]
@dsnipe Thanks, I've just merged it! :+1:

**Comment by jodosha** [comment]
Closed by #172
